.github/workflows/analyze.yml

```diff
@@ -0,0 +1,33 @@
+name: "Analyze"
+on:
+  push:
+    branches:
+      - main
+      - release/**
+  pull_request:
+    paths-ignore:
+      - "**/*.md"
+
+jobs:
+  cancel-previous-workflow:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@01ce38bf961b4e243a6342cbade0dbc8ba3f0432 # pin@0.12.0
+        with:
+          access_token: ${{ github.token }}
+
+  analyze:
+    runs-on: macos-latest
```

Comment:
also, I think this could run on ubuntu, doesn't need any apple stuff right? I think ubuntu is faster?